x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -183,7 +175,7 @@ protected LogicalPlan rule(Filter filter) {
                         if (cmp instanceof Equals) {
                             result = new IsNull(cmp.source(), comparableToNull);
                         } else {
-                            result = new IsNotNull(cmp.source(), comparableToNull);
+                            result = new Not(cmp.source(), new IsNull(cmp.source(), comparableToNull));
```

Comment:
I would have guessed that `IsNotNull` translates more natively to a query DSL `exists` but that `IsNull` was just the negation of that.

Reply:
The presence of `not` enforces the right behavior in case of multi-value handling, on the idea that **all** value should not match, thus the negation of a non-negated condition. Regarding query DSL, yes, it's probably equivalent to `exists`. I'll leave that for a follow up investigation and/or improvement. Atm, `field != null` translates into a `must_not [must_not [exists]]` which is _probably_ equivalent to `exists`.